client/src/modules/authorization/components/SignInForm.jsx

```diff
@@ -0,0 +1,185 @@
+import React, { PureComponent } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import validator from 'validator';
+import { connect } from 'react-redux';
+import _flowRight from 'lodash/flowRight';
+
+import AuthInput from './AuthInput';
+import { signIn } from 'modules/authorization/model/actions';
+import PendingButton from 'common/components/PendingButton';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import { UnauthorizedException } from 'common/exceptions/UnauthorizedException';
+import { IntlPropType } from 'common/constants/propTypes';
+
+class SignInForm extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    email: '',
+    isEmailValid: false,
+    isFormValid: false,
+    isPasswordValid: false,
+    password: '',
+    pending: false,
+    signUpErrorId: ''
+  };
+
+  componentDidUpdate() {
+    this.isFormValid();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  onEmailChange = (email, isValid) => {
+    const { signUpErrorId } = this.state;
+    const error = isValid ? '' : signUpErrorId;
+    this.setState({
+      email,
+      isEmailValid: isValid,
+      signUpErrorId: error
+    });
+  };
+
+  onPasswordChange = (password, isValid) => {
+    const { signUpErrorId } = this.state;
+    const error = isValid ? '' : signUpErrorId;
+    this.setState(
+      {
+        password,
+        isPasswordValid: isValid,
+        signUpErrorId: error
+      },
+      this.comparePasswords
+    );
+  };
+
+  emailValidator = value => {
+    const { isEmail } = validator;
+
+    if (!isEmail(value)) {
+      return 'authorization.input.email.invalid';
+    }
+
+    return '';
+  };
+
+  isFormValid = () => {
+    const { isEmailValid, isPasswordValid, signUpErrorId } = this.state;
+
+    return this.setState({
+      isFormValid: isEmailValid && isPasswordValid && !signUpErrorId
+    });
+  };
+
+  handleSignIn = () => {
+    const { email, password } = this.state;
+    const { signIn } = this.props;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(signIn(email, password));
+
+    return this.pendingPromise.promise
+      .then(() => this.setState({ pending: true }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const newState = { pending: false };
+
+          if (err instanceof UnauthorizedException) {
+            newState.signUpErrorId =
+              'authorization.actions.sign-in.invalid-credentials';
+          } else {
```

Comment:
else not needed here

Reply:
`else` is needed because there is no return in `if` statement. If I remove `else` signInErrorId would always be 'common.something-went-wrong'

Reply:
so you can add 'return' at the end

Reply:
1. If we add return in if statement we will get the eslint error "Return statement should not contain assignment."
2. Return inside this if statement will exit catch statement